userService.js

```diff
@@ -0,0 +1,130 @@
+const mssql = require("mssql/msnodesqlv8");
+const dbConfig = require("./dbconfig");
+const sqlHelper = require("./userServiceSQLHelper");
+
+const pool = new mssql.ConnectionPool(dbConfig.config);
+
+function paginateUsers(req, res) {
+    pool.connect(() => {
+        var sql = "SELECT * FROM dbo.Users";
+        pool.query(sql, function(err, result) {
+            var page = req;
+            var limit = 5;
+            var startIndex = (page - 1) * limit;
+            var endIndex = page * limit;
+            var resultUsers = result.recordset.slice(startIndex, endIndex);
+            res.send(resultUsers);
+        });
+        mssql.close();
+    });
+};
+
+function getUsers(req, res) {
+    var sortBy = sqlHelper.sortBy(req.sorting);
+    var sortDirection = sqlHelper.sortDirection(req.order);
+    pool.connect().then(() => {
+        if (req.filter) {
+            var sql = `
+            SELECT *
+            FROM dbo.Users
+            WHERE FirstName LIKE @Search OR LastName LIKE @Search OR Age LIKE @Search OR Phone LIKE @Search OR Email LIKE @Search OR Gender LIKE @Search
+            ORDER BY ${sortBy} ${sortDirection}
```

Comment:
[SQL injection is possible ](https://www.imperva.com/learn/application-security/sql-injection-sqli/#:~:text=SQL%20injection%2C%20also%20known%20as,lists%20or%20private%20customer%20details.)

Reply:
Could you please clarify what do you mean?
I tried moving the ${sortBy} and ${sortDirection} to @sortBy and @sortDirection, and adding the input() after with necessary info. But the app just crashes.